src/components/main-view/main-view.jsx

```diff
@@ -19,8 +23,6 @@ export class MainView extends React.Component {
   constructor() {
     super();
     this.state = {
-      movies: [],
-      user: null,
       hasAccount: true
```

Comment:
OK.  Yes `hasAccount` is to move between login and registration with the `handleRegister` and `handleReturnLogin` methods. When it's false, registration-view is displayed.  This is different to `user === null` because that moves between login and the main-view.  I'll change it from state to redux.